tests/karate/src/test/java/be/election/castVote.feature

```diff
@@ -14,70 +14,170 @@ Feature: Cast a vote
+          "created_at": 1633098941,
+          "votes": [
+            {
+              "id": '#(getInvalidVoteId)',
+              "question": '#(getIsThisProjectFunQuestionId)',
+              "vote": [0]
+            }
+          ]
+        }
+      """
+    When frontend.publish(JSON.stringify(invalidCastVote), electionChannel)
+    And json answer = frontend.getBackendResponse(JSON.stringify(invalidCastVote))
+    Then match answer contains INVALID_MESSAGE_FIELD
+    And match frontend.receiveNoMoreResponses() == true
   # Testing if after creating an election correctly, the backend returns an error
-  # upon casting a valid vote but after the election end time
-  Scenario: Casting a valid vote too late should return an error
-    Given string badCastVoteData = read('classpath:data/election/data/castVote/bad_cast_vote_late_vote_data.json')
-    And string badCastVote = converter.publishМessageFromData(badCastVoteData, castVoteId, electionChannel)
-    * call read('classpath:be/utils/simpleScenarios.feature@name=election_setup')
-    And eval frontend.send(badCastVote)
-    * json cast_vote = frontend_buffer.takeTimeout(timeout)
-    Then match cast_vote contains deep {jsonrpc: '2.0', id: '#(castVoteId)', error: {code: -4, description: '#string'}}
-
+  # upon casting a vote but with lao id as vote id
+  Scenario: Casting a vote with lao id as vote id should return an error
+    Given call read('classpath:be/utils/simpleScenarios.feature@name=election_setup')
+    And def invalidCastVote =
+      """
+        {
+          "object": "election",
+          "action": "cast_vote",
+          "lao": '#(getLaoValid)',
+          "election": '#(getValidElectionSetupId)',
+          "created_at": 1633098941,
+          "votes": [
+            {
+              "id": '#(getLaoValid)',
+              "question": '#(getIsThisProjectFunQuestionId)',
+              "vote": [0]
+            }
+          ]
+        }
+      """
+    When frontend.publish(JSON.stringify(invalidCastVote), electionChannel)
+    And json answer = frontend.getBackendResponse(JSON.stringify(invalidCastVote))
+    Then match answer contains INVALID_MESSAGE_FIELD
+    And match frontend.receiveNoMoreResponses() == true
+#  # Testing if after creating an election correctly, the backend returns an error
+#  # upon casting a valid vote but after the election end time
+#  Scenario: Casting a valid vote too late should return an error
+#    Given string badCastVoteData = read('classpath:data/election/data/castVote/bad_cast_vote_late_vote_data.json')
+#    And string badCastVote = converter.publishМessageFromData(badCastVoteData, castVoteId, electionChannel)
+#    * call read('classpath:be/utils/simpleScenarios.feature@name=election_setup')
+#    And eval frontend.send(badCastVote)
+#    * json cast_vote = frontend_buffer.takeTimeout(timeout)
+#    Then match cast_vote contains deep {jsonrpc: '2.0', id: '#(castVoteId)', error: {code: -4, description: '#string'}}
+#
```

Comment:
Should this be merged with master ?

Reply:
Probably not, I was wondering ow to adapt the casting a late vote into the new specifications of the protocol that's why I left it there